### PR TITLE
Singleton Examples

### PR DESCRIPTION
**Pull request by duane701**
Please review and merge into master. 

These changes provide an example for constructing subsystem and command singletons.

### CODE REVIEW

File: src/OI.cpp [hunk 1]
```diff
@@ -3,16 +3,46 @@
...
+, coA(coDriver.get(), kButtonA_ID)
+, coB(coDriver.get(), kButtonB_ID)
+, coX(coDriver.get(), kButtonX_ID)
+, coY(coDriver.get(), kButtonY_ID)
+, coLB(coDriver.get(), kButtonLB_ID)
+, coRB(coDriver.get(), kButtonRB_ID)
+, coL3(coDriver.get(), kButtonL3_ID)
+, coR3(coDriver.get(), kButtonR3_ID)
+, coStart(coDriver.get(), kButtonStart_ID)
+, coBack(coDriver.get(), kButtonBack_ID)
```
Comment (OliviaRS) [inline src/OI.cpp:38]: We usually space out codriver and driver controls so they're easier to see

File: src/OI.h [hunk 2]
```diff
@@ -6,38 +6,42 @@
 class OI {
 private:
 
+  static std::shared_ptr<OI> self;
+
 //..........driver..........
 
   std::shared_ptr<Joystick> driver;
```
Comment (OliviaRS) [inline src/OI.h:13]: Have we looked into that xbox controller class?

**Review by OliviaRS — COMMENTED** [review]
LGTM

other than the consistent naming conventions that I mentioned earlier

**Review by Diego-Andrade — APPROVED** [review]
LGTM Merging

**Pull request merged**